### ISSUE TITLE
Document issues with mail locks on MySQL < 5.7.5 in docs/requirements.md

### ISSUE THREAD

**Issue by MegaphoneJon** [civicrm/civicrm-sysadmin-guide#158]
See [Mattermost chat](https://chat.civicrm.org/civicrm/pl/ftknqojwzbrp3m6i535qwj55cr) for details.

**Comment by mattwire** [comment]
And mariadb < 10.0.2

See https://github.com/civicrm/civicrm-core/pull/13854

**Comment by mattwire** [comment]
My suggestion is to add a check to CiviCRM as well and warn if using an older mysql version

**Comment by mfb** [comment]
Yes, documenting this is needed.  For the record, I'd also be in favor of fixing the bug by changing how locks work - but I don't want to work on a PR for that unless there is some chance of it being accepted :)

**Comment by mfb** [comment]
Also, if we don't fix Civi core, then we could add a link to this module as a work-around: https://www.drupal.org/project/civicrm_lock (I didn't make an official release of it yet, but I can if there's interest).

**Comment by MikeyMJCO** [comment]
Does this need further action in light of the merged core PR linked by Matt?

**Comment by eileenmcnaughton** [comment]
I guess the new advice is to run civicrm on mysql 5.7.5+ or equivalent MariaDB - I think we should say those are our minimum recommended versions

**Comment by mfb** [comment]
This bothers me because it means older versions are "supported" but significantly broken.  It seems more honest to say they're not supported?

**Comment by eileenmcnaughton** [comment]
it looks like mysql 5.7.5 was released on 2014-09-25 - I think we could call it min version if it's 5 years old!

**Comment by MikeyMJCO** [comment]
I'm inclined to agree that we update the "officially supported range with a clear note about why those older versions are "unsupported"?"

**Comment by mfb** [comment]
At some point soon it would be nice to take advantage of MySQL 5.7+ features, meaning that older versions would be both not supported and not compatible. But I guess, for the moment, if older versions sorta work but are not supported it's helpful to explain why.

**Comment by JoeMurray** [comment]
Could we include equivalent versions for Percona as well as MariaDB? 

@mfb too bad civicrm_lock is a Drupal module rather than civi extension.

**Comment by mfb** [comment]
> @mfb too bad civicrm_lock is a Drupal module rather than civi extension.

I made it a drupal module because it relies on Drupal 7; if someone makes a Civi extension it would make sense to have it create the lock table itself rather than using Drupal's semaphore table.

**Comment by eileenmcnaughton** [comment]
Argh I meant to create a PR but  looks like I straight committed - https://github.com/civicrm/civicrm-sysadmin-guide/commit/5a7f296432dcd31eb1ce4a72c2f96d4c3e47e44b

This is why I don't have merge rights on docs repos!

**Comment by MikeyMJCO** [comment]
:-D

**Comment by mfb** [comment]
Seems like the CiviCRM UI itself now needs some install and status warnings for older versions of MySQL?

**Comment by MikeyMJCO** [comment]
I'd agree - open an issue against core on GL maybe?
